app/views/main/AllServicesOn.tsx

```diff
@@ -1,25 +1,81 @@
-import React from 'react';
-import { Dimensions, ImageBackground, StatusBar, View } from 'react-native';
+import React, { useState, useEffect, useCallback } from 'react';
+import {
+  Dimensions,
+  ImageBackground,
+  StatusBar,
+  View,
+  TouchableOpacity,
+} from 'react-native';
 import { useTranslation } from 'react-i18next';
 import Pulse from 'react-native-pulse';
 import { SvgXml } from 'react-native-svg';
+import Geolocation from '@react-native-community/geolocation';
 
 import { Icons, Images } from '../../assets';
 import { Typography } from '../../components/Typography';
 
 import { styles } from './style';
 import { Colors } from '../../styles';
+import { Stacks, NavigationProp } from '../../navigation';
+import { useSelector, useDispatch } from 'react-redux';
+import isAutoSubscriptionEnabledSelector from '../../store/selectors/isAutoSubscriptionEnabledSelector';
+import selectedHealthcareAuthoritiesSelector from '../../store/selectors/selectedHealthcareAuthoritiesSelector';
+import getHealthcareAuthorities from '../../store/actions/healthcareAuthorities/getHealthcareAuthoritiesAction';
 
 type AllServicesOnProps = {
   noHaAvailable: boolean;
+  navigation: NavigationProp;
+};
+
+const useAutoSubscriptionBanner = (
+  showAutoSubscribeBanner: boolean,
+  navigation: NavigationProp,
+) => {
+  const [showBanner, setShowBanner] = useState(showAutoSubscribeBanner);
+
+  useEffect(() => {
+    const listener = navigation.addListener('blur', () => setShowBanner(false));
+
+    return listener.remove;
+  }, [navigation]);
+
+  return {
+    showBanner,
+  };
 };
 
 export const AllServicesOnScreen = ({
   noHaAvailable,
+  navigation,
 }: AllServicesOnProps): JSX.Element => {
+  const size = Dimensions.get('window').height;
   const { t } = useTranslation();
+  const dispatch = useDispatch();
 
-  const size = Dimensions.get('window').height;
+  const { showBanner } = useAutoSubscriptionBanner(true, navigation);
+  const autoSubscriptionEnabled = useSelector(
+    isAutoSubscriptionEnabledSelector,
+  );
+  const selectedAuthorities = useSelector(
+    selectedHealthcareAuthoritiesSelector,
+  );
+
+  const [haName, setHAName] = useState('');
+  const autoSubscribe = useCallback(async () => {
+    if (autoSubscriptionEnabled && noHaAvailable) {
+      Geolocation.getCurrentPosition(({ coords }) => {
+        dispatch(getHealthcareAuthorities(undefined, coords));
+      });
+    }
+  }, [autoSubscriptionEnabled, noHaAvailable, dispatch]);
+
+  useEffect(() => {
+    autoSubscribe().then(() => {
+      if (selectedAuthorities[0]) {
+        setHAName(selectedAuthorities[0].name);
```

Comment:
I don't think this callback will act as expected because this is dependent on the callback chain of thunks resolving. Also, I don't think the 
```js
   const [haName, setHAName] = useState(''); 
``` 
is needed. 
Instead, we can directly use the selector value. 
```js
const haName = useSelector(state => state.selectedAuthorities[0]?.name)
```

Reply:
fixed!